common/storage.go

```diff
@@ -0,0 +1,178 @@
+package common
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/giantswarm/clustertest/pkg/client"
+	"github.com/giantswarm/clustertest/pkg/wait"
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var (
+	pvcName    = "pvc-test"
+	pvcPodName = "pvc-test-pod"
+)
+
+func runStorage() {
+	Context("storage", func() {
+		var wcClient *client.Client
+
+		BeforeEach(func() {
+			var err error
+
+			wcClient, err = Framework.WC(Cluster.Name)
+			if err != nil {
+				Fail(err.Error())
+			}
+		})
+
+		It("has a at least one storage class available", func() {
+			Eventually(wait.Consistent(checkStorageClassExists(wcClient), 10, time.Second)).
+				WithTimeout(wait.DefaultTimeout).
+				WithPolling(wait.DefaultInterval).
+				Should(Succeed())
+		})
+
+		It("has created a pod with a pvc and the pvc is bound", func() {
+			Eventually(wait.Consistent(createPodWithPVC(wcClient), 10, time.Second)).
+				WithTimeout(wait.DefaultTimeout).
+				WithPolling(wait.DefaultInterval).
+				Should(Succeed())
+		})
+
+		It("has deleted a pod with a pvc", func() {
+			Eventually(wait.Consistent(deletePodWithPVC(wcClient), 10, time.Second)).
+				WithTimeout(wait.DefaultTimeout).
+				WithPolling(wait.DefaultInterval).
+				Should(Succeed())
+		})
+	})
+}
+
+func checkStorageClassExists(wcClient *client.Client) func() error {
+	return func() error {
+		// ensure we have at least one storage class available
+		storageClasses := &storagev1.StorageClassList{}
+		err := wcClient.List(context.Background(), storageClasses)
+		if err != nil {
+			return err
+		}
+		if len(storageClasses.Items) == 0 {
+			return fmt.Errorf("no storage classes found")
+		}
+
+		return nil
+	}
+}
+
+func createPodWithPVC(wcClient *client.Client) func() error {
```

Comment:
Nitpick. It would probably be nicer to have these manifests in yaml in an `assets` directory. In general I think we should have an test app that we deploy on the cluster once that uses all the features like PVC that we want to test, but we can move in that direction later.